File: Integer-break/integerbreak.py
import math
class Solution :
    def integerBreak(self, n: int) -> int:
        
        rem1 = n % 3 
        y = n//3
        if n ==1 or n==2 :
            return 1 
        elif n==3:
            return 2
        elif rem1 ==1 :
            s = pow(3,y-1)*4
            return s
        elif rem1==0:
            return pow(3,y)
        else:
            s = pow(3,y)*2
            return s 
              
        



        

File: Integer-break/Integer-break.cpp
#define max
class Solution
{
public:
    int integerBreak(int n)
    {
        if (n < 2)
        {
            return 0;
        }
        int dp[n + 1];
        dp[1] = 1;
        for (int i = 2; i <= n; i++)
        {
            dp[i] = 0;
            for (int j = 1; j < i; j++)
            {
                dp[i] = max(dp[i], max(j * (i - j), j * dp[i - j]));
            }
        }
        return dp[n];
    }
};